### ISSUE TITLE
Add a dedicated "Troubleshooting" guide

### ISSUE THREAD

**Issue by clason** [nvim-treesitter/nvim-treesitter#2301]
Tree-sitter integration is a complex thing, with many moving parts, and users are frequently at a loss what the issue is when things go wrong.

We should create a dedicated `Troubleshooting.md` that we can link from the issue template and point people in chat towards. This should be more structured than the current section in the README, giving step-by-step instructions ("first check this; if that happens, do this; try turning it off and on again").

I recommend we start small and extend it continuously whenever things pop up in discussion (like the recent non-standard parser installation directory), so a first draft should not be daunting -- making this a good first issue for someone looking to contribute!

**Comment by vigoux** [comment]
I think having it in the wiki would be nice, to allow for the community to contribute to it easily, and simply the process of adding sections to it.

**Comment by clason** [comment]
Maybe; I thought having it as an actual file would be more visible, though, and allow for more quality control.

It's not a bad option, though!

**Comment by vigoux** [comment]
Indeed...
Thinking about it more, using an actual file may actually be better, and as you say, allow more quality control.

**Comment by MunifTanjim** [comment]
Does it makes sense to ship a semi-automated troubleshooter? e.g., for automatically finding the broken query files:

![Peek 2022-01-20 21-34](https://user-images.githubusercontent.com/8050659/150370403-be02e55d-3c77-4a8d-b8d4-a703feedf8ae.gif)

**Comment by MunifTanjim** [comment]
On second thought, these things can actually go into `:checkhealth nvim-treesitter` 🤔

**Comment by clason** [comment]
Yeah, exposing more information in `:checkhealth` would good regardless (especially parser location, revision, etc.). But this is less about information and more about what to do; basically a list of things to try first _before_ opening an issue. (E.g., all the different directories one should delete to get back to "factory defaults".)
